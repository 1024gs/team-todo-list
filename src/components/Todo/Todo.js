import React from 'react';
import './Todo.scss';
import TodoHeader from '../TodoHeader/TodoHeader';
import TodoList from '../TodoList/TodoList';
import TodoAdd from '../TodoAdd/TodoAdd';

const Todo = ({items, group, getItemId, onAdd, onEdit, onToggle, onDelete}) => {
  return (
    <div className="Todo">
      <TodoHeader
        title={group.title}
        subtitle={new Date().toLocaleString('en-GB', {
          weekday: 'short',
          day: 'numeric',
          month: 'long',
          timeZone: 'UTC'
        })}
      />
      <TodoList
        items={items}
        getItemId={getItemId}
        onEdit={onEdit}
        onToggle={onToggle}
        onDelete={onDelete}
      />
      <TodoAdd onAdd={onAdd} />
    </div>
  );
};

export default Todo;
