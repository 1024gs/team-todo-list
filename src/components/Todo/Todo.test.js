import React from 'react';
import Todo from './Todo';
import sinon from 'sinon';
import {shallow} from 'enzyme';

const setup = (propOverrides) => {
  const props = Object.assign(
    {
      items: [
        {id: 1, title: 'Todo 1'},
        {id: 2, title: 'Todo 2', isCompleted: true}
      ],
      group: {title: 'Team To-do List'},
      getItemId: (x) => x.id,
      onAdd: sinon.spy(),
      onToggle: sinon.spy(),
      onDelete: sinon.spy()
    },
    propOverrides
  );
  const wrapper = shallow(<Todo {...props} />);

  return {
    props,
    wrapper
  };
};

describe('Todo', () => {
  it('renders', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
