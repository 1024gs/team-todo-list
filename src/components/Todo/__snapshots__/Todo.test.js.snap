// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders 1`] = `
<div
  className="Todo"
>
  <TodoHeader
    subtitle="Sun, November 10"
    title="Team To-do List"
  />
  <TodoList
    getItemId={[Function]}
    items={
      Array [
        Object {
          "id": 1,
          "title": "Todo 1",
        },
        Object {
          "id": 2,
          "isCompleted": true,
          "title": "Todo 2",
        },
      ]
    }
    onDelete={[Function]}
    onToggle={[Function]}
  />
  <TodoAdd
    onAdd={[Function]}
  />
</div>
`;
